packages/quick_actions/quick_actions_ios/ios/Classes/FLTQuickActionsPlugin.m

```diff
@@ -76,9 +82,10 @@ - (BOOL)application:(UIApplication *)application
 }
 
 - (void)applicationDidBecomeActive:(UIApplication *)application {
-  if (self.launchingShortcutType) {
-    [self handleShortcut:self.launchingShortcutType];
-    self.launchingShortcutType = nil;
+  NSString *shortcut = self.shortcutStateManager.launchingShortcutType;
+  if (shortcut) {
+    [self handleShortcut:shortcut];
+    self.shortcutStateManager.launchingShortcutType = nil;
```

Comment:
Can you explain more why you are separating out a `FLTShortcutStateManager`?  Everywhere you use it now looks like feature envy, which makes me think they should be one class, as they are now.  I can see making the functions into class methods, if that's easier to test.

Reply:
The main reason is that I want to directly inject `launchingShortcutType` so testing `applicationDidBecomeActive` is quick and easy. 

Alternative approach is to set this `launchingShortcutType` via `didFinishLaunchingWithOptions` public API, and keep `launchingShortcutType` a private ivar of `FLTQuickActionsPlugin`. But this will likely require 2-3 additional tests to reach similar level of confidence. I think I will go with the second approach here then.

Reply:
Updated the code.